internal/validator/validator.go

```diff
@@ -0,0 +1,302 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+// Package validator provides components to create epoch claims and update
+// rollups outputs with their proofs.
+package validator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/cartesi/rollups-node/internal/merkle"
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// The maximum height for the Merkle tree of all outputs produced
+// by an application
+const MAX_OUTPUT_TREE_HEIGHT = 63
+
+type ValidatorRepository interface {
+	// GetAllRunningApplications returns a slice with the applications currently
+	// being validated by the node.
+	GetAllRunningApplications(ctx context.Context) ([]Application, error)
```

Comment:
The way repository interface is partitioned makes this method be declared more than once. This could lead to some inconsistencies. 

But as a repository redesign will take place in the newer future we can prevent this pointing out a single doc source for some methods